search/segment_tree.hpp

```diff
@@ -64,11 +64,30 @@ class SegmentTree
   void Add(Segment const & segment);
   void Erase(Segment const & segment);
 
-  // Calls |fn| on arbitrary segment containing |x|.
+  // Calls |fn| on an arbitrary segment containing |x|.
   template <typename Fn>
-  void Find(double x, Fn && fn) const
+  void FindAny(double x, Fn && fn) const
   {
-    return Find(0 /* index */, x, fn);
+    FindAny(0 /* index */, x, fn);
+  }
+
+  // Calls |fn| on all segments containing |x| making
+  // exactly one call per segment.
+  template <typename Fn>
+  void FindAll(double x, Fn && fn)
```

Comment:
Looks good now, did you notice any speedups?

Reply:
~4x over the previous version
~150x over the original